google-beta/resource_accesscontextmanager_access_policy.go

```diff
@@ -0,0 +1,315 @@
+
+	// The operation for this resource contains the generated name that we need
+	// in order to perform a READ. We need to access the object inside of it as
+	// a map[string]interface, so let's do that.
+	bytes, err := op.Response.MarshalJSON()
+	if err != nil {
+		return err
+	}
+
+	var data map[string]interface{}
+	if err := json.Unmarshal(bytes, &data); err != nil {
+		return err
+	}
+
+	name := GetResourceNameFromSelfLink(data["name"].(string))
+	log.Printf("[DEBUG] Setting AccessPolicy name, id to %s", name)
+	d.Set("name", name)
+	d.SetId(name)
+
+	return resourceAccessContextManagerAccessPolicyRead(d, meta)
+}
+
+func resourceAccessContextManagerAccessPolicyRead(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+
+	url, err := replaceVars(d, config, "https://accesscontextmanager.googleapis.com/v1beta/accessPolicies/{{name}}")
+	if err != nil {
+		return err
+	}
+
+	res, err := sendRequest(config, "GET", url, nil)
+	if err != nil {
+		return handleNotFoundError(err, d, fmt.Sprintf("AccessContextManagerAccessPolicy %q", d.Id()))
+	}
+
+	if err := d.Set("name", flattenAccessContextManagerAccessPolicyName(res["name"])); err != nil {
+		return fmt.Errorf("Error reading AccessPolicy: %s", err)
+	}
+	if err := d.Set("create_time", flattenAccessContextManagerAccessPolicyCreateTime(res["createTime"])); err != nil {
+		return fmt.Errorf("Error reading AccessPolicy: %s", err)
+	}
+	if err := d.Set("update_time", flattenAccessContextManagerAccessPolicyUpdateTime(res["updateTime"])); err != nil {
+		return fmt.Errorf("Error reading AccessPolicy: %s", err)
+	}
+	if err := d.Set("parent", flattenAccessContextManagerAccessPolicyParent(res["parent"])); err != nil {
+		return fmt.Errorf("Error reading AccessPolicy: %s", err)
+	}
+	if err := d.Set("title", flattenAccessContextManagerAccessPolicyTitle(res["title"])); err != nil {
+		return fmt.Errorf("Error reading AccessPolicy: %s", err)
+	}
+
+	return nil
+}
+
+func resourceAccessContextManagerAccessPolicyUpdate(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+
+	obj := make(map[string]interface{})
+	parentProp, err := expandAccessContextManagerAccessPolicyParent(d.Get("parent"), d, config)
+	if err != nil {
+		return err
+	} else if v, ok := d.GetOkExists("parent"); !isEmptyValue(reflect.ValueOf(v)) && (ok || !reflect.DeepEqual(v, parentProp)) {
+		obj["parent"] = parentProp
+	}
+	titleProp, err := expandAccessContextManagerAccessPolicyTitle(d.Get("title"), d, config)
+	if err != nil {
+		return err
+	} else if v, ok := d.GetOkExists("title"); !isEmptyValue(reflect.ValueOf(v)) && (ok || !reflect.DeepEqual(v, titleProp)) {
+		obj["title"] = titleProp
+	}
+
+	url, err := replaceVars(d, config, "https://accesscontextmanager.googleapis.com/v1beta/accessPolicies/{{name}}")
+	if err != nil {
+		return err
+	}
+
+	log.Printf("[DEBUG] Updating AccessPolicy %q: %#v", d.Id(), obj)
+	updateMask := []string{}
+
+	if d.HasChange("parent") {
```

Comment:
The comments at https://cloud.google.com/access-context-manager/docs/reference/rest/v1beta/accessPolicies make me think that parent can't be changed. Have you tried, by any chance?

Reply:
It's can't! It's `input` in MM, so it's `ForceNew` in the schema. This is an existing issue; see [Filestore Instance `networks`](https://github.com/terraform-providers/terraform-provider-google-beta/blob/2.0.0/google-beta/resource_filestore_instance.go#L297) as well.

I tried changing this by filtering `input` fields, and it's a little more nuanced than I'd hoped; afaik (based on testing like... 3 resources) it works as intended for resources that use `PATCH`, but it breaks some resources that use `PUT`. In the scope of this PR, I'd like to cite prior art since this won't have any effect on users, and I'll work out a separate PR to clean up the fields we send on `Update` as a followup